src/openms/source/CHEMISTRY/EnzymaticDigestion.cpp

```diff
@@ -171,20 +160,56 @@ namespace OpenMS
     log_model_threshold_ = threshold;
   }
 
+  std::vector<Size> EnzymaticDigestion::tokenise(const AASequence& protein) const
+  {
+    String s = protein.toUnmodifiedString();
+    boost::regex re(enzyme_.getRegEx());
+    boost::sregex_token_iterator i(s.begin(), s.end(), re, -1);
+    boost::sregex_token_iterator j;
+    std::vector<Size> positions;
+    Size pos = -1;
+    while (i != j)
+    {
+      AASequence pep = AASequence::fromString((String)(*i++));
```

Comment:
this is relativly slow (as an AASequence object needs to be constructed). It should be possible to do directly use i->size() as the position. (The underlying sub_match ( see http://www.boost.org/doc/libs/1_34_0/libs/regex/doc/sub_match.html ) is a std::string.)